pontoon/base/models.py

```diff
@@ -2913,30 +2933,94 @@ def calculate_stats(self, save=True):
         # Singular
         translations = Translation.objects.filter(
             entity__in=translated_entities.filter(string_plural=''),
-            locale=locale
+            locale=locale,
+        ).prefetch_related(
+            'errors',
+            'warnings',
         )
-        approved = translations.filter(approved=True).count()
-        fuzzy = translations.filter(fuzzy=True).count()
-        missing = resource.total_strings - approved - fuzzy
+
+        approved = translations.filter(
```

Comment:
[nit] Maybe it's a good moment to add new method, e.g. `Translation.objects.without_errors()`?

Reply:
Possibly, I'll see if it's needed while I fix bug 1486606.